x-pack/plugins/apm/server/lib/service_nodes/index.ts

```diff
@@ -83,6 +93,7 @@ const getServiceNodes = async ({
       name: bucket.key as string,
       cpu: bucket.cpu.value,
       heapMemory: bucket.heapMemory.value,
+      hostName: bucket.latest.top[0].metrics['host.hostname'],
```

Comment:
What's the type of this? Elsewhere I saw you are expecting it to be `string | number | undefined`:

![image](https://user-images.githubusercontent.com/209966/130457950-6dbbf417-bb22-46de-b4c8-50573c8a3dd6.png)


I don't think this can ever be number so perhaps cast it to:
```suggestion
      hostName: bucket.latest.top[0].metrics['host.hostname'] as string | undefined,
```

Reply:
Is it possible that `bucket.latest.top[0]` returns undefined?  if yes it will fail when you try to get `metrics`.

Reply:
Host name can be returned as null.
But I'm going to get rid of the function, as I created thinking in more complicated cases, than just print value or not

Reply:
Good point. Yes, we should guard against that.

Reply:
I think this might be enough:
`hostName: bucket.latest.top[0] &&
        (bucket.latest.top[0].metrics['host.hostname'] as
          | string
          | null
          | undefined),`
 What do you think? @sqren @cauemarcondes 
 
 Btw: Is not github suppose to resolve conversations automatically when the file is updated?

Reply:
I think you could do something like this.

```js
hostname: bucket.latest.top?.[0]?.metrics?.['host.hostname']
```

PS: I didn't test it 🙈

> Btw: Is not github suppose to resolve conversations automatically when the file is updated?

Not really, you have to do it manually.